bin/sdl_run

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
+##############################################################################
+#
+#
+# Line Segment Tracking Standalone Code Run Script
+#
+#
+##############################################################################
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+# Help
+usage()
+{
+  echo "ERROR - Usage:"
+  echo
+  echo "      sh $(basename $0) OPTIONSTRINGS ..."
+  echo
+  echo "Options:"
+  echo "  -h    Help                    (Display this message)"
+  echo "  -f    compilation flags       (Compilation flags e.g. mc NOTE: Must start with '-' e.g. '-mc')"
+  echo "  -s    sample name             (Sample name e.g. PU200, muonGun, ...)"
+  echo "  -n    number of events        (Number of events to run over)"
+  echo "  -t    tag for this run        (Tag for this run)"
+  echo
+  exit
+}
+
+# Parsing command-line opts
+while getopts ":f:s:n:t:h" OPTION; do
+  case $OPTION in
+    f) FLAGS=${OPTARG};;
+    s) SAMPLE=${OPTARG};;
+    n) NEVENTS=${OPTARG};;
+    t) TAG=${OPTARG};;
+    h) usage;;
+    :) usage;;
+  esac
+done
+
+# If the command line options are not provided set it to default value of false
+if [ -z ${FLAGS} ]; then usage; fi
+if [ -z ${SAMPLE} ]; then usage; fi
+if [ -z ${NEVENTS} ]; then NEVENTS=-1; fi
+if [ -z ${TAG} ]; then usage; fi
+
+# Check that the FLAGS start with "-" character
+if [[ ${FLAGS:0:1} == "-" ]]; then
+    :
+else
+    echo "ERROR:"
+    echo ""
+    echo "Compilation flag Option provided is '-f ${FLAGS}'"
+    echo "However, compilation flags must start with '-' e.g. '-mc'"
+    echo ""
+    exit
+fi
+
+# Shift away the parsed options
+shift $(($OPTIND - 1))
+
+# Move to the TRACKLOOPERDIR
+pushd ${TRACKLOOPERDIR}
+
+# Verbose
+echo "====================================================="
+echo "Line Segment Tracking Run Script                     "
+echo "====================================================="
+echo ""
+echo "  COMPILATION FLAG  : ${FLAGS}"
+echo "  SAMPLE            : ${SAMPLE}"
+echo "  NEVENTS           : ${NEVENTS}"
+echo "  TAG               : ${TAG}"
+echo ""
+echo "  (cf. Run > sh $(basename $0) -h to see all options)"
+echo ""
+
+JOBTAG=${TAG}_${SAMPLE}_NEVT${NEVENTS}
+
+rm -f ${LSTOUTPUTDIR}/${JOBTAG}__LSTRun.log
```

Comment:
Running the same command for this script repeatedly fails because all of the scripts used within it stop running if their output files previously exist. Can we add a flag, say `-d` that would take care of deleting all these files, such as `rm ./testRun_PU200_NEVT100__LSTNtuple.root ./testRun_PU200_NEVT100__LSTNumDen.root`

By the way, is it easy to put the outputs in a folder called `${JOBTAG}`? Otherwise the top directory gets very quickly cluttered...

Reply:
The output dir can be changed via,

export LSTOUTPUTDIR=/my/favorite/output/dir

I will think about the existing output... thingy..
I usually don't like overwriting something that's already written.
But even if that's the case I guess a better "fail exit messagE" would be better.

Reply:
> The output dir can be changed via,
> `export LSTOUTPUTDIR=/my/favorite/output/dir`

Sure. But let's have the default something that works.

Reply:
> I will think about the existing output... thingy..
> I usually don't like overwriting something that's already written.
> But even if that's the case I guess a better "fail exit messagE" would be better.

I agree and that's why I propose a flag that the user would consciously put there. The issue with the error messages is that the failure happens in multiple stages.

In my case, I had to do
```
rm ./testRun_PU200_NEVT100__LSTNtuple.root
```
only to get another fail for the existence of `./testRun_PU200_NEVT100__LSTNumDen.root`, so did
```
rm ./testRun_PU200_NEVT100__LSTNumDen.root
```
only to get another error, because I needed to do both removals at once. I imagine this could get very tiring, very quickly...

Reply:
I added mkdir -p 
and also added file override features